### DISCUSSION TITLE
icy-title tag in ogg container?

### DISCUSSION THREAD

**Discussion by artur-shaik** [savonet/liquidsoap#3585] [Q&A] [answered]
Is it possible somehow to have `icy-title` tag in ogg container?

Here is my output configuration:
```
output.icecast(
    %ogg(%vorbis(quality=0.8,samplerate=44100,channels=2)),
    mount="/high.ogg",
    send_icy_metadata=true,
    host=icecast_host, port=icecast_port, user="highsrc", password=high_pass,
    radio)
```

and here is ffprobe results:
```
ffprobe -show_streams -show_format https://ice.coldstar.online/high.ogg
...
Input #0, ogg, from 'https://ice.coldstar.online/high.ogg':
  Metadata:
    icy-index-metadata: 1
    icy-logo        : https://coldstar.online/static/assets/firefox_app_512x512.png
    icy-country-code: kg
    icy-language-codes: en,ru
    icy-main-stream-url: https://ice.coldstar.online/high.ogg
    icy-genre       : trance,house
    icy-description : This stream contains classics of progressive/psy/uplifting trance as well as tech house and techno. Radioshows with cues, and live tracklist on ColdStar.Online. Ads free!
    icy-name        : ColdStar.Online Radio (High quality)
    icy-pub         : 1
  Duration: N/A, start: 5552.966667, bitrate: 256 kb/s
  Stream #0:0: Audio: vorbis, 48000 Hz, stereo, fltp, 256 kb/s
    Metadata:
      title           : Solarstone - Pure Trance Radio 384 (2023-11-29)
      server          : Icecast 2.4.4
...
```

so, here is no `icy-title` tag, this leads to the fact that all android apps cannot display current playing metadata, because all of them using [ExoPlayer](https://github.com/androidx/media) library which relay on that tag.

**Comment by vitoyucepi** [comment]
Hi @artur-shaik,
I don't know about exoplayer, but the structure of ogg streams is different from mp3 streams.
1. MPEG streams send `StreamTitle` header every `icy-metaint` bytes if `icy-metadata: 1` header is set.
1. `icy-title` and other http headers are received by the client only once at the beginning of the connection. To update them, players must probe the server every *N* seconds.
1. OGG streams don't support `icy-metadata` header.
1. OGG streams always transfer headers and metadata inside the container.

**Reply by artur-shaik** [reply]
Thank you for information.

**Comment by artur-shaik** [comment, marked as answer]
Actually, this is a bug in android library, which should be [fixed](https://github.com/androidx/media/issues/876) in next release of ExoPlayer, I guess.
